src/test/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoProcessorIT.java

```diff
@@ -6,20 +6,42 @@
 package org.opensearch.geospatial.ip2geo.processor;
 
 import java.io.IOException;
+import java.time.Duration;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import lombok.SneakyThrows;
 
 import org.opensearch.client.Response;
 import org.opensearch.client.ResponseException;
+import org.opensearch.common.Randomness;
 import org.opensearch.geospatial.GeospatialRestTestCase;
 import org.opensearch.geospatial.GeospatialTestHelper;
+import org.opensearch.geospatial.ip2geo.action.PutDatasourceRequest;
+import org.opensearch.geospatial.ip2geo.common.DatasourceState;
 
 public class Ip2GeoProcessorIT extends GeospatialRestTestCase {
+    /**
+     * This is sample manifest file pointing to a sample csv data
+     * The sample data contains three lines as follows
+     *
+     * cidr,city,country
+     * 10.0.0.0/8,Seattle,USA
+     * 127.0.0.0/12,Vancouver,Canada
+     * fd12:2345:6789:1::/64,Bengaluru,India
+     *
+     */
+    private static final String SAMPLE_ENDPOINT = "https://geoip.maps.opensearch.org/v1/sample/city/manifest.json";
```

Comment:
Does it mean the integration test have to access the endpoint to run successfully? If yes, I have a concern if the IT need to run air-gapped environment.

Reply:
I think the Integration test result need to be independent from server side status

Reply:
Why do we need to run integ test in air-gapped environment?

Reply:
It's a open source plugin, the use environment may be variable by different client. Can we directly mock the manifest response in the plugin?

Reply:
Mocking does not work for integ test. Let me find a way to launch file server locally.